client/lib/wp/sync-handler/index.js

```diff
@@ -121,26 +163,31 @@ export class SyncHandler {
 			 * WP.com server-side response.
 			 *
 			 * @param {Object} serverResponse - server response object
+			 * @returns {Promise} - promise that resolves with server response object
 			 */
 			const cacheResponse = serverResponse => {
-				// get response object without _headers property
-				let responseWithoutHeaders = Object.assign( {}, serverResponse );
-				delete responseWithoutHeaders._headers;
-
-				let storingData = {
-					__sync: {
-						key,
-						synced: Date.now(),
-						syncing: false
-					},
-					body: responseWithoutHeaders,
-					params: reqParams
-				};
-
-				// add/override gotten data from server-side
-				this.storeRecord( key, storingData ).catch( err => {
-					// @TODO error handling
-					warn( err );
+				return new Promise( ( resolve ) => {
```

Comment:
I wasn't sure where to put `checkPagination` other than at the end of the `.then()` chain, and doesn't the prior `then()` (`cacheResponse` here) need to be a promise at that point? It doesn't really need `cacheResponse` to resolve first though. I didn't want to call it directly from `cacheResponse` because I want to keep this functionally separate. Can you suggest a better way?